woodwork/logical_types.py

```diff
@@ -71,7 +72,16 @@ def transform(self, series):
         return series
 
     def validate(self, series, return_invalid_values=True):
-        return
+        """Validates that a logical type is consistent with the series dtype. Performs additional type
+        specific validation, as required."""
+        valid_dtype = self._get_valid_dtype(type(series))
+        if valid_dtype != str(series.dtype):
```

Comment:
I like the idea of this as the default check! 

Just want to make sure there wasn't a specific reason @jeff-hernandez went the simpler route originally. My one concern is that the other validation checks specific to the logical types are checking the actual data values, so a dtype check that fails won't have individual rows that fail to be returned when `return_invalid_values=True`. Do we just say all values are invalid?

Reply:
Commented to use super() for the logical type is not ordinal nor latlong at this moment

Reply:
I had brought this up in [this comment](https://github.com/alteryx/woodwork/pull/1247#discussion_r785197462) during the implementation. I wasn't sure if we were already checking the dtypes elsewhere. If we are checking this somewhere else, then we might want to remove it and only keep this one to avoid redundant checks.

Reply:
As part of `LogicalType.transform`, we change the dtype to match if needed. But if a user did something to their dataframe that changed the dtype without removing woodwork types it would, in theory invalidate the logical type! So I like the idea of having it called in `LogicalType.validate`. We just need to make sure the outputs can work for `return_invalid_values` since right now that's just being ignored.

Reply:
Talked with @jeff-hernandez and decided that we like the current behavior! 

The reason being that if the dtype doesn't match the logical type it indicates a change by the user that invalidated the logical type, so we should always error, even if `return_invalid_values` is True. 

Let's add a comment at this method explaining why the dtype error will still show up when `return_invalid_values` is True, though.